overlord/fdestate/backend/reseal.go

```diff
@@ -100,93 +103,79 @@ func getUniqueModels(bootChains []boot.BootChain) []secboot.ModelForSealing {
 	return models
 }
 
-func resealKeyForBootChainsFDEHook(method device.SealingMethod, rootdir string, inputs resealInputs, opts resealOptions) error {
-	params := inputs.bootChains
+func doReseal(manager FDEStateManager, method device.SealingMethod, rootdir string) error {
+	_, runModels, runPCRProfile, err := manager.Get("run+recover", "all")
+	if err != nil {
+		return err
+	}
+
+	_, recoveryModels, recoveryPCRProfile, err := manager.Get("recover", "system-save")
```

Comment:
gadget has constants for things like "system-save", I even wonder if we should give them a type

Reply:
Yes we should. I think we have FIXMEs on the Set calls.

Reply:
we haven't been great about adding types to various constants used in the code, but a bunch of calls here take string arguments right next to each other, so even a simple swap error wouldn't be properly caught right now